src/framework/global/thirdparty/haw_logger/logger/logger.cpp

```diff
@@ -61,7 +61,7 @@ DateTime DateTime::now()
     dt.time.hour = tm->tm_hour;
     dt.time.min = tm->tm_min;
     dt.time.sec = tm->tm_sec;
-    dt.time.msec = static_cast<int>(ms_d.count() - (sec * 1000));
+    dt.time.msec = static_cast<int>(ms_d.count() - (static_cast<long long>(sec) * 1000));
```

Comment:
Whoudn't `dt.time.msec = static_cast<int>(ms_d.count() - (sec * 1000LL));` be better? As the latter is the culprit for the potential overflow.

Reply:
done